src/tree.rs

```diff
@@ -629,11 +629,14 @@ impl<T> Tree<T> {
     ///
     /// Swaps two `Node`s including their children given their `NodeId`s.
     ///
-    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    /// If one `Node` is a descendant of the other getting swapped, the former *upper* `Node` is attached as last child of the former *lower* `Node` after the swap. (The *lower* is taking the *uppers* original position as usual.)
+    /// The subtree of the former *upper* node is not touched except for the *lower* node moved including all its children.
+    ///
```

Comment:
Could you change this:
"...is not touched except for the *lower* node moved including..."
to:
"...is not touched except that the *lower* node is moved including..."